### PR TITLE
feat: PBI 16 cron api assertions andrew

### PR DESCRIPTION
**Pull request by andrew-susanto**
### Describe your changes
PBI-16 cron api assertions

### Backlog name
MON65: [BE] edit cron API assertion

### Acceptance criteria
- User can add expected result (json, text) on Create new API monitor, edit API monitor page
- API Monitor can assert response of each API and give error if failed to assert

### Checklist
- [x] I have performed a self-review of my code
- [x] Code successfully run on local
- [x] Feature / changes tested on local
- [x] No failing unit test
- [ ] Passed UAT Test
- [x] Sonarqube tested 
- [ ] Required any changes to environment variable

### CODE REVIEW

File: cron/management/commands/run_cron.py [hunk 1]
```diff
@@ -77,6 +78,58 @@ def replace_string_with_json_result(self, text, dict):
...
         return text
+
+    def run_api_monitor_assertions(self, monitor_id, response):
+        monitor = APIMonitor.objects.get(id=monitor_id)
+        if monitor.assetion_type == 'TEXT' and response != monitor.assertion_value:
+            raise AssertionError('Assertion text failed')
+        elif monitor.assetion_type == 'JSON':
+            try:
+                api_response = json.loads(response) 
+            except json.decoder.JSONDecodeError as e:
```
Comment (luthfifahlevi) [inline cron/management/commands/run_cron.py:89]: `as e` mungkin di hapus aja

File: cron/management/commands/run_cron.py [hunk 2]
```diff
@@ -77,6 +78,58 @@ def replace_string_with_json_result(self, text, dict):
...
+                        key_part = key_part.replace(array_idx[0], '')
+                        res_key += f"['{key_part}']{array_idx[0]}"
+                    else:
+                        res_key += f"['{key_part}']"
+                ddiff_exclude_path.append(res_key)
+            
+            ddiff = DeepDiff(assertion_value, api_response, exclude_paths=ddiff_exclude_path)
+            
+            diff_result = ""
+            if not monitor.is_assert_json_schema_only:
```
Comment (luthfifahlevi) [inline cron/management/commands/run_cron.py:115]: fungsi ini mungkin bisa di refactor lagi? jdi 2 atau 3 function. Karena sekarang jdi code smell:
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=42&id=MonAPI-xyz_MonAPI&open=AYPqSM6_bXBUHiW4rtzL

File: cron/management/commands/run_cron.py [hunk 3]
```diff
@@ -77,6 +78,58 @@ def replace_string_with_json_result(self, text, dict):
...
+            if 'iterable_item_added' in ddiff:
+                diff_result += f"Found iterable item added with keys {ddiff['iterable_item_added'].keys()}\n"
+            if 'iterable_item_removed' in ddiff:
+                diff_result += f"Found iterable item removed with keys {ddiff['iterable_item_removed'].keys()}\n"
+
+            if diff_result != "":
+                raise AssertionError(diff_result.strip('\n'))
             
     def run_api_monitor_request(self, monitor_id, monitor_history):
         monitor_history.append(monitor_id)
```
Comment (luthfifahlevi) [inline cron/management/commands/run_cron.py:135]: Fungsi ini juga mungkin direfactor lagi hehe thank you dan semangat ndrew
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=42&id=MonAPI-xyz_MonAPI&open=AYPqSM6_bXBUHiW4rtzM

File: cron/management/commands/run_cron.py (hunk 2 above)
Comment (andrew-susanto) [reply cron/management/commands/run_cron.py:115]: hmmm wkkwkwk, agak sulit yaa soalnya kalo diliat2 codenya lebih enak single func

**Review by ferdifdi — APPROVED** [review]

**Pull request merged**